packages/@uppy/compressor/src/index.js

```diff
@@ -52,15 +52,22 @@ export default class Compressor extends BasePlugin {
           this.uppy.log(`[Image Compressor] Image ${file.id} compressed by ${prettierBytes(compressedSavingsSize)}`)
           totalCompressedSize += compressedSavingsSize
           const { name, type, size } = compressedBlob
-          const extension = name && getFileNameAndExtension(name).extension
+          const { extension } = name && getFileNameAndExtension(name)
+          // Name (file.meta.name) might have been changed by user, so we update only the extension
+          const newMetaName = file.meta.name && extension && `${getFileNameAndExtension(file.meta.name).name}.${extension}`
```

Comment:
- only set metadata if it's actually defined, rather than `undefined.jpg` (where `extension` is defined but meta name isn't
- made it a bit more clear with naming why we have to call `getFileNameAndExtension` twice and always call `getFileNameAndExtension`, where it just returns `undefined` if it's not defined.

```suggestion
          const compressedFileName = getFileNameAndExtension(name)
          const metaFileName = getFileNameAndExtension(file.meta.name)
          // The meta name might have been changed by user, if that's the case we need to update the extension.
          if (metaFileName) {
            const newName = `${metaFileName.name}.${compressedFileName.extension}`
            this.uppy.setFileMeta({ name })
          }
```